eth_account/signers/ledger.py

```diff
@@ -0,0 +1,362 @@
+
+import eth_utils
+import rlp
+
+from hexbytes import HexBytes
+
+from eth_account import (
+    Account,
+)
+
+from eth_account.datastructures import (
+    AttributeDict,
+)
+
+from eth_account.signers.base import (
+    BaseAccount,
+)
+
+from eth_account.internal.transactions import (
+    encode_transaction,
+    serializable_unsigned_transaction_from_dict,
+)
+
+from eth_utils import (
+    to_bytes,
+    to_hex,
+)
+
+from eth_utils.curried import (
+    keccak,
+)
+
+
+ETH_DERIVATION_PATH_PREFIX = "m/44'/60'/0'/"
+
+CHANNEL_ID = 0x0101
+TAG_APDU = 0x05
+TAG_PING = 0x02
+
+# Packet HEADER is defined at
+# https://github.com/LedgerHQ/blue-app-eth/blob/master/doc/ethapp.asc#general-transport-description
+#
+# 1. Communication channel ID (big endian) value is always CHANNEL_ID
+# 2. Command tag (TAG_APDU or TAG_PING)
+# 3. Packet sequence index (big endian) starting at 0x00
+#
+PACKET_HEADER = struct.pack('>HBH', CHANNEL_ID, TAG_APDU, 0x00)
+PACKET_SIZE = 64  # in bytes
+PACKET_FREE = PACKET_SIZE - len(PACKET_HEADER)
+
+
+def wrap_apdu(command):
+    '''
+    Return a list of packet to be sent to the device
+    '''
+    packets = []
+
+    # Prefix command with its length
+    command = struct.pack('>H', len(command)) + command
+
+    # Split command into at max PACKET_FREE sized chunks
+    commands = [command[i:i + PACKET_FREE] for i in range(0, len(command), PACKET_FREE)]
+
+    # Create a packet for each command chunk
+    for packet_id in range(len(commands)):
+        header = struct.pack('>HBH', CHANNEL_ID, TAG_APDU, packet_id)
+        packet = header + commands[packet_id]
+
+        # Add padding to the packet to make it exactly PACKET_SIZE long
+        packet += bytes([0x0] * (PACKET_SIZE - len(packet)))
+
+        packets.append(packet)
+
+    return packets
+
+
+def unwrap_apdu(packet):
+    '''
+    Given a packet from the device, extract and return relevant info
+    '''
+    if not packet:
+        return (None, None, None, None, None)
+
+    (channel, tag, packet_id, reply_size) = struct.unpack('>HBHH', packet[:7])
+
+    if packet_id == 0:
+        # reply_size is only valid in first reply
+        return (channel, tag, packet_id, reply_size, packet[7:])
+    else:
+        return (channel, tag, packet_id, None, packet[5:])
+
+
+class LedgerUsbException(Exception):
+    pass
+
+
+class LedgerUsbDevice:
+    '''
+    References:
+    - https://github.com/LedgerHQ/blue-loader-python/blob/master/ledgerblue/comm.py#L56
+    - https://github.com/ethereum/go-ethereum/blob/master/accounts/usbwallet/ledger.go
+    '''
+
+    logger = logging.getLogger('eth_account.signers.ledger.LedgerUsbDevices')
+
+    def __init__(self):
+        hidDevicePath = None
+        for hidDevice in hid.enumerate(0, 0):
+            if hidDevice['vendor_id'] == 0x2c97:
+                if ('interface_number' in hidDevice and hidDevice['interface_number'] == 0) \
+                   or ('usage_page' in hidDevice and hidDevice['usage_page'] == 0xffa0):
```

Comment:
Could these hex values be put in global constants?  `LEDGER_VENDOR_ID` and `LEDGER_USAGE_PAGE_ID`?